presto-main/src/main/java/com/facebook/presto/operator/Operator.java

```diff
@@ -60,6 +60,30 @@
     Page getOutput();
 
     /**
+     * After calling this method operator should revoke all reserved revocable memory.
+     * As soon as memory is revoked returned future should be marked as done.
+     * <p>
+     * Spawned threads can not modify OperatorContext because it's not thread safe.
+     * For this purpose use implement finishMemoryRevoke
+     * <p>
+     * After startMemoryRevoke is called on Operator the Driver is disallowed to call any
+     * processing methods on it (finish/isFinished/isBlocked/needsInput/addInput/getOutput) until
+     * finishMemoryRevoke is called.
+     */
+    default ListenableFuture<?> startMemoryRevoke()
+    {
+        return NOT_BLOCKED;
```

Comment:
This is wrong default implementation. This method is called when `Operator` _has_ some revocable memory and should release it. Default implementation should be:
```
throw new UnsupportedOperationException(format("%s allocates revocable memory but does not implement revoking", getClass()));
```